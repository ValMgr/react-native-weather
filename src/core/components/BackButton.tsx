import React, { useCallback } from 'react';
import { View, Button } from 'react-native';
import { useNavigation } from '@react-navigation/native';

export const BackButton = () => {
  const navigation = useNavigation();

  const handlePress = useCallback(() => {
    navigation.goBack();
  }, [navigation]);

  return (
    <View>
      <Button title='Back' onPress={handlePress} />
    </View>
  );
};
