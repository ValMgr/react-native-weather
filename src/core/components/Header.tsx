import { View, Text, StyleSheet } from 'react-native';

export const Header = () => {
  return (
    <View style={styles.header} pointerEvents={'none'}>
      <Text style={styles.headerText}>.Weather</Text>
      {/* <Text style={styles.headerText}>v0.1.0</Text> */}
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-end',
    width: '100%',
    paddingTop: 55,
    padding: 20,
    position: 'absolute',
    top: 0,
    left: 0,
    zIndex: 99,

  },
  headerText: {
    color: '#BB86FC',
    fontSize: 20,
    fontWeight: 'bold',
  },
});
