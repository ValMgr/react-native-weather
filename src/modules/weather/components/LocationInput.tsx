import React, { useCallback } from 'react';
import { View, TextInput, Button, StyleSheet, NativeSyntheticEvent, TextInputChangeEventData } from 'react-native';
import { useNavigation } from '@react-navigation/native';

import { geocodeCity } from '@core/services/api';
import { useWeather } from '@modules/weather/providers/WeatherProvider';

type Nav = {
  navigate: (screen: string) => void;
};

export const LocationInput = () => {
  const { setLocation, city, setCity } = useWeather();
  const navigation = useNavigation<Nav>();

  const handleLocationChange = useCallback(
    (e: NativeSyntheticEvent<TextInputChangeEventData>) => {
      setCity(e.nativeEvent.text);
    },
    [setCity],
  );

  const handleLocationSubmit = useCallback(() => {
    if (city) {
      geocodeCity(city).then((location) => {
        setLocation(location);
        navigation.navigate('Weather');
      });
    }
  }, [city]);

  return (
    <View style={styles.inputContainer}>
      <TextInput style={styles.input} onChange={handleLocationChange} placeholder='Enter a location' />
      <Button color={'#BB86FC'} title='Search' onPress={handleLocationSubmit} />
    </View>
  );
};

const styles = StyleSheet.create({
  inputContainer: {
    display: 'flex',
    flexDirection: 'row',
    gap: 10,
    width: '80%',
    marginHorizontal: '10%',
    marginVertical: 20,
  },
  input: {
    backgroundColor: '#EFEFEF',
    borderRadius: 10,
    borderWidth: 1,
    borderColor: '#BB86FC',
    font: '400 18px Roboto',
    padding: 10,
    flex: 1,
  },
});
