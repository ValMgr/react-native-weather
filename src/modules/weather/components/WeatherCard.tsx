import { useMemo } from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';

import { getWeatherIcon, getDay } from '@modules/weather/services/weather_display';

interface IProps {
  day: number;
  temperature_max: number;
  temperature_min: number;
  weather_code: number;
}

export const WeatherCard = ({ day, temperature_max, temperature_min, weather_code }: IProps) => {
  const icon = useMemo(() => getWeatherIcon(weather_code), [weather_code]);
  const dayName = useMemo(() => getDay(day), [day]);

  return (
    <View style={styles.card}>
      <Image style={styles.weatherIcon} source={{uri: icon }} />
      <Text style={styles.day}>{dayName}</Text>
      <View style={styles.temperatureContainer}>
        <Text style={styles.temperature}>{Math.round(temperature_max)}°</Text>
        <Text style={styles.temperature}>{Math.round(temperature_min)}°</Text>
      </View>
    </View>
  );

};

const styles = StyleSheet.create({
  card: {
    display: 'flex',
    flexDirection: 'row',
    gap: 10,
    alignItems: 'center',
    backgroundColor: '#FFFFFF',
    borderRadius: 10,
    padding: 10,
    margin: 10,
  },
  weatherIcon: {
    width: 50,
    height: 50,
  },
  day: {
    fontSize: 20,
    fontWeight: 'bold',
    marginHorizontal: 10
  },
  temperatureContainer: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-end',
    flex: 1,
  },
  temperature: {
    fontSize: 20,
    fontWeight: 'bold',
  },

});
