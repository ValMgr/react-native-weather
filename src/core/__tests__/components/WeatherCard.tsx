import renderer from 'react-test-renderer';

import { WeatherCard } from '@modules/weather/components/WeatherCard';

const props = {
  day: 0,
  temperature_max: 22,
  temperature_min: 12,
  weather_code: 0,
};

test('WeatherCard renders correctly', () => {
  const tree = renderer.create(<WeatherCard {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});