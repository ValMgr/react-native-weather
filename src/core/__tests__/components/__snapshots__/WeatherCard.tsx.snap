// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "display": "flex",
      "flexDirection": "row",
      "gap": 10,
      "margin": 10,
      "padding": 10,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://help.apple.com/assets/6222428998C2CE34C75A5252/6222428B98C2CE34C75A5267/fr_FR/35c778528f79f2aa038b07526447f606.png",
      }
    }
    style={
      {
        "height": 50,
        "width": 50,
      }
    }
  />
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    Aujourd'hui
  </Text>
  <View
    style={
      {
        "alignItems": "flex-end",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      22
      °
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      12
      °
    </Text>
  </View>
</View>
`;
